src/lib/network/network_autoyast.rb

```diff
@@ -104,26 +106,40 @@ def set_network_service
       NetworkService.EnableDisableNow
     end
 
-    # Initializates DNS setup according AY profile
+    # Initializates NICs setup according AY profile
     #
-    # FIXME: it currently doesn't write DNS configuration. It is used for initialization
-    # of DNS setup according AY profile in 1st stage as part of network setup was moved
-    # here already and some parts of network configuration needs to know it. DNS write
-    # is still done in 2nd stage.
+    # If the installer is running in 1st stage mode only, then the configuration
+    # is also written
     #
+    # @param [Boolean] write forces instant writing of the configuration
     # @return [Boolean] true when configuration was present and loaded from the profile
-    def configure_dns
-      ay_dns_config = ay_networking_section["dns"]
+    def configure_lan(write: false)
+      log.info("NetworkAutoYast: Lan configuration")
 
-      return false if !ay_dns_config
+      ay_configuration = Lan.FromAY(ay_networking_section)
+      ay_configuration = NetworkAutoYast.instance.merge_configs(ay_configuration) if keep_net_config?
 
-      DNS.Import(ay_dns_config)
+      configure_submodule(Lan, ay_configuration, write: write)
+    end
 
-      log.info("NetworkAutoYast: DNS / Hostname configuration")
-      log.info("dhcp hostname: #{DNS.dhcp_hostname}")
-      log.info("write hostname: #{DNS.write_hostname}")
+    # Initializates /etc/hosts according AY profile
+    #
+    # If the installer is running in 1st stage mode only, then the configuration
+    # is also written
+    #
+    # @param [Boolean] write forces instant writing of the configuration
+    # @return [Boolean] true when configuration was present and loaded from the profile
+    def configure_hosts(write: false)
+      log.info("NetworkAutoYast: Hosts configuration")
 
-      true
+      # expected format for Hosts.Import is { "ip" => [list, of, names] }
```

Comment:
OK, and this must be documented *there*, in the API docs for Host.Import (not Hosts).